spark-commons-test/src/main/scala/za/co/absa/spark/commons/test/JDKCollectionConvertersCompat.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2021 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.spark.commons.test
+
+// Solution based on conversation in scala-collection-compat
+// https://github.com/scala/scala-collection-compat/issues/208
+object JDKCollectionConvertersCompat {
```

Comment:
While this is pretty clever it's still really hard to understand how it works.
Wouldn't it be better to use shims like [spark-hats](https://github.com/AbsaOSS/spark-hats/tree/master/src/main) or [slick ](https://github.com/slick/slick/tree/v3.3.3/slick/src/main)use.

Reply:
Happy to go either way. I just used a way they did it in [scala-collection-compat](https://github.com/scala/scala-collection-compat) but at the same time, I tried avoiding adding a new dependency.